src/main/java/org/jenkinsci/plugins/workflow/log/TaskListenerDecorator.java

```diff
@@ -70,7 +70,7 @@
     private static final long serialVersionUID = 1;
 
     private static final Logger LOGGER = Logger.getLogger(TaskListenerDecorator.class.getName());
-
+    private static final boolean useReverseOrder=Boolean.getBoolean(TaskListenerDecorator.class.getName()+".useReverseOrder");
```

Comment:
This is still switching the behavior for all implementations and doing so by default.

FTR my suggestion was to introduce a method in `TaskListenerDecorator.Factory` so that the behavior would be unmodified by default; the different order would be available as an option to a particular implementation which wished to work that way.

Reply:
I think we need to understand why the `reverse()` call was there and is it reasonable? if yes, we should not modify the default behavior.

I don't think introducing new method in TaskListenerDecorator.Factory will solve the issue because in the code 
```
Stream.concat(ExtensionList.lookup(TaskListenerDecorator.Factory.class).stream().map(f -> f.of(owner)), Stream.of(mainDecorator))
```
the order of `mainDecorator` was the cause

Reply:
Which is why `apply` would look something like

```java

        List<TaskListenerDecorator> decorators = Stream.concat(
                ExtensionList.lookup(TaskListenerDecorator.Factory.class).stream().filter(TaskListenerDecorator.Factory::reversed.negate()).map(f -> f.of(owner)),
                Stream.of(mainDecorator),
                ExtensionList.lookup(TaskListenerDecorator.Factory.class).stream().filter(TaskListenerDecorator.Factory::reversed).map(f -> f.of(owner))).
```

Reply:
thanks, updated PR